std/file.d

```diff
@@ -3088,11 +3093,12 @@ else version(Posix)
             _dTypeSet = false;
         }
 
-        private this(string path, core.sys.posix.dirent.dirent* fd)
+        private this(string path, core.sys.posix.dirent.dirent* fd) @trusted
         {
             import std.path : buildPath;
+            import std.algorithm.searching : countUntil;
 
-            immutable len = core.stdc.string.strlen(fd.d_name.ptr);
+            immutable len = fd.d_name[].countUntil(0);
```

Comment:
I'll create a bug report and and a PR to fix this shortly, but using `countUntil` instead of `strlen` is completely incorrect here. I don't know why that was done, but the decoding that `countUntil` does is completely unnecessary, and counting code points and then slicing on code units is fundamentally wrong, and it means that this code no longer correctly handles file names with non-ASCII characters. Their names get truncated.